src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -41,8 +41,203 @@
 #include "rest/ConnectionInfo.h"
 #include "rest/uriParamNames.h"
 
+#include "mongo/client/dbclient.h"
+#include "mongoBackend/dbConstants.h"
+#include "mongoBackend/dbFieldEncoding.h"
+
+using namespace mongo;
 using namespace orion;
 
+/* ****************************************************************************
+*
+* Forward declarations
+*/
+static void compoundValueBson(std::vector<CompoundValueNode*> children, BSONObjBuilder& b);
+
+
+
+/* ****************************************************************************
+*
+* compoundValueBson (for arrays) -
+*/
+static void compoundValueBson(std::vector<CompoundValueNode*> children, BSONArrayBuilder& b)
+{
+  for (unsigned int ix = 0; ix < children.size(); ++ix)
+  {
+    CompoundValueNode* child = children[ix];
+
+    if (child->valueType == ValueTypeString)
+    {
+      b.append(child->stringValue);
+    }
+    else if (child->valueType == ValueTypeNumber)
+    {
+      b.append(child->numberValue);
+    }
+    else if (child->valueType == ValueTypeBoolean)
+    {
+      b.append(child->boolValue);
+    }
+    else if (child->valueType == ValueTypeNone)
+    {
+      b.appendNull();
+    }
+    else if (child->valueType == ValueTypeVector)
+    {
+      BSONArrayBuilder ba;
+
+      compoundValueBson(child->childV, ba);
+      b.append(ba.arr());
+    }
+    else if (child->valueType == ValueTypeObject)
+    {
+      BSONObjBuilder bo;
+
+      compoundValueBson(child->childV, bo);
+      b.append(bo.obj());
+    }
+    else
+    {
+      LM_E(("Runtime Error (Unknown type in compound value)"));
```

Comment:
It shouldn't, but I'd like to keep the LM_E in the else just in case some bug (e.g. some change in the if/else if chain) causes the program flow end in that part. Just loging a Runtime Error (so we can detect in running instances) would be enough.

NTC